### ISSUE TITLE
error: unrecognized command line option ‘-std=c++20’; did you mean ‘-std=c++2a’?

### ISSUE THREAD

**Issue by LittleYang0531** [K0lb3/UnityPy#268]
**Code**
https://github.com/K0lb3/UnityPy/blob/b4bc2ce500d702db16fd5c34309adde0fa651762/setup.py#L72

**Error**
```bash
Building wheels for collected packages: UnityPy
  Building wheel for UnityPy (PEP 517): started
  Building wheel for UnityPy (PEP 517): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python /tmp/tmpjfik8q8i build_wheel /tmp/tmpjkh9zhzt
Failed to build UnityPy
       cwd: /tmp/pip-req-build-iivqgc10
  Complete output (163 lines):
  running bdist_wheel
  running build
  running build_py
  creating build/lib.linux-x86_64-cpython-38/UnityPy
  copying UnityPy/environment.py -> build/lib.linux-x86_64-cpython-38/UnityPy
  copying UnityPy/exceptions.py -> build/lib.linux-x86_64-cpython-38/UnityPy
  copying UnityPy/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy
  copying UnityPy/config.py -> build/lib.linux-x86_64-cpython-38/UnityPy
  creating build/lib.linux-x[86](https://github.com/SonolusHaniwa/sirius-sync/actions/runs/11009695039/job/30569914874#step:5:87)_64-cpython-38/UnityPy/math
  copying UnityPy/math/Color.py -> build/lib.linux-x86_64-cpython-38/UnityPy/math
  copying UnityPy/math/Half.py -> build/lib.linux-x86_64-cpython-38/UnityPy/math
  copying UnityPy/math/Vector3.py -> build/lib.linux-x86_64-cpython-38/UnityPy/math
  copying UnityPy/math/Vector4.py -> build/lib.linux-x86_64-cpython-38/UnityPy/math
  copying UnityPy/math/Quaternion.py -> build/lib.linux-x86_64-cpython-38/UnityPy/math
  copying UnityPy/math/Vector2.py -> build/lib.linux-x86_64-cpython-38/UnityPy/math
  copying UnityPy/math/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/math
  copying UnityPy/math/Matrix4x4.py -> build/lib.linux-x86_64-cpython-38/UnityPy/math
  copying UnityPy/math/Rectangle.py -> build/lib.linux-x86_64-cpython-38/UnityPy/math
  creating build/lib.linux-x86_64-cpython-38/UnityPy/export
  copying UnityPy/export/SpriteHelper.py -> build/lib.linux-x86_64-cpython-38/UnityPy/export
  copying UnityPy/export/MeshRendererExporter.py -> build/lib.linux-x86_64-cpython-38/UnityPy/export
  copying UnityPy/export/ShaderConverter.py -> build/lib.linux-x86_64-cpython-38/UnityPy/export
  copying UnityPy/export/Texture2DConverter.py -> build/lib.linux-x86_64-cpython-38/UnityPy/export
  copying UnityPy/export/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/export
  copying UnityPy/export/AudioClipConverter.py -> build/lib.linux-x86_64-cpython-38/UnityPy/export
  copying UnityPy/export/MeshExporter.py -> build/lib.linux-x86_64-cpython-38/UnityPy/export
  creating build/lib.linux-x86_64-cpython-38/UnityPy/classes
  copying UnityPy/classes/PPtr.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes
  copying UnityPy/classes/Object.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes
  copying UnityPy/classes/generated.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes
  copying UnityPy/classes/math.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes
  copying UnityPy/classes/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes
  creating build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/BundleFile.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/FileType.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/ShaderGpuProgramType.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/ShaderCompilerPlatform.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/SpriteMeshType.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/ClassIDType.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/BuildTarget.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/SpritePackingRotation.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/VertexFormat.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/Audio.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/PassType.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/TextureFormat.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/CommonString.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/GraphicsFormat.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/GfxPrimitiveType.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/ExtendableEnum.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/SpritePackingMode.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/SerializedPropertyType.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/MeshTopology.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  copying UnityPy/enums/TextureDimension.py -> build/lib.linux-x86_64-cpython-38/UnityPy/enums
  creating build/lib.linux-x86_64-cpython-38/UnityPy/lib
  copying UnityPy/lib/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/lib
  creating build/lib.linux-x86_64-cpython-38/UnityPy/files
  copying UnityPy/files/ObjectReader.py -> build/lib.linux-x86_64-cpython-38/UnityPy/files
  copying UnityPy/files/BundleFile.py -> build/lib.linux-x86_64-cpython-38/UnityPy/files
  copying UnityPy/files/File.py -> build/lib.linux-x86_64-cpython-38/UnityPy/files
  copying UnityPy/files/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/files
  copying UnityPy/files/SerializedFile.py -> build/lib.linux-x86_64-cpython-38/UnityPy/files
  copying UnityPy/files/WebFile.py -> build/lib.linux-x86_64-cpython-38/UnityPy/files
  creating build/lib.linux-x86_64-cpython-38/UnityPy/resources
  copying UnityPy/resources/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/resources
  creating build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/TypeTreeNode.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/ImportHelper.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/TypeTreeHelper.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/MeshHelper.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/PackedBitVector.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/ResourceReader.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/TextureSwizzler.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/ArchiveStorageManager.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/Tpk.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  copying UnityPy/helpers/CompressionHelper.py -> build/lib.linux-x86_64-cpython-38/UnityPy/helpers
  creating build/lib.linux-x86_64-cpython-38/UnityPy/streams
  copying UnityPy/streams/EndianBinaryReader.py -> build/lib.linux-x86_64-cpython-38/UnityPy/streams
  copying UnityPy/streams/EndianBinaryWriter.py -> build/lib.linux-x86_64-cpython-38/UnityPy/streams
  copying UnityPy/streams/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/streams
  creating build/lib.linux-x86_64-cpython-38/UnityPy/classes/legacy_patch
  copying UnityPy/classes/legacy_patch/Shader.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes/legacy_patch
  copying UnityPy/classes/legacy_patch/Mesh.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes/legacy_patch
  copying UnityPy/classes/legacy_patch/Renderer.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes/legacy_patch
  copying UnityPy/classes/legacy_patch/AudioClip.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes/legacy_patch
  copying UnityPy/classes/legacy_patch/Texture2DArray.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes/legacy_patch
  copying UnityPy/classes/legacy_patch/Sprite.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes/legacy_patch
  copying UnityPy/classes/legacy_patch/Texture2D.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes/legacy_patch
  copying UnityPy/classes/legacy_patch/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/classes/legacy_patch
  creating build/lib.linux-x86_64-cpython-38/UnityPy/lib/FMOD
  copying UnityPy/lib/FMOD/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/lib/FMOD
  copying UnityPy/resources/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/resources
  creating build/lib.linux-x86_64-cpython-38/UnityPy/tools
  copying UnityPy/tools/extractor.py -> build/lib.linux-x86_64-cpython-38/UnityPy/tools
  creating build/lib.linux-x86_64-cpython-38/UnityPy/tools/libil2cpp_helper
  copying UnityPy/tools/libil2cpp_helper/helper.py -> build/lib.linux-x86_64-cpython-38/UnityPy/tools/libil2cpp_helper
  copying UnityPy/tools/libil2cpp_helper/il2cpp_class.py -> build/lib.linux-x86_64-cpython-38/UnityPy/tools/libil2cpp_helper
  copying UnityPy/tools/libil2cpp_helper/il2cpp.py -> build/lib.linux-x86_64-cpython-38/UnityPy/tools/libil2cpp_helper
  copying UnityPy/tools/libil2cpp_helper/metadata.py -> build/lib.linux-x86_64-cpython-38/UnityPy/tools/libil2cpp_helper
  copying UnityPy/tools/libil2cpp_helper/__init__.py -> build/lib.linux-x86_64-cpython-38/UnityPy/tools/libil2cpp_helper
  copying UnityPy/tools/libil2cpp_helper/metadata_class.py -> build/lib.linux-x86_64-cpython-38/UnityPy/tools/libil2cpp_helper
  running egg_info
  writing UnityPy.egg-info/PKG-INFO
  writing dependency_links to UnityPy.egg-info/dependency_links.txt
  writing requirements to UnityPy.egg-info/requires.txt
  writing top-level names to UnityPy.egg-info/top_level.txt
  reading manifest file 'UnityPy.egg-info/SOURCES.txt'
  adding license file 'LICENSE'
  writing manifest file 'UnityPy.egg-info/SOURCES.txt'
  copying UnityPy/resources/uncompressed.tpk -> build/lib.linux-x86_64-cpython-38/UnityPy/resources
  creating build/lib.linux-x86_64-cpython-38/UnityPy/lib/FMOD/Linux/x86_64
  copying UnityPy/lib/FMOD/Linux/x86_64/libfmod.so -> build/lib.linux-x86_64-cpython-38/UnityPy/lib/FMOD/Linux/x86_64
  copying UnityPy/resources/uncompressed.tpk -> build/lib.linux-x86_64-cpython-38/UnityPy/resources
  copying UnityPy/lib/FMOD/Linux/x86_64/libfmod.so -> build/lib.linux-x86_64-cpython-38/UnityPy/lib/FMOD/Linux/x86_64
  copying UnityPy/resources/uncompressed.tpk -> build/lib.linux-x86_64-cpython-38/UnityPy/resources
  running build_ext
  building 'UnityPy.UnityPyBoost' extension
  creating build/temp.linux-x86_64-cpython-38/UnityPyBoost
  x86_64-linux-gnu-g++ -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/tmp/pip-req-build-iivqgc10/UnityPyBoost -I/usr/include/python3.8 -c UnityPyBoost/AnimationClip.cpp -o build/temp.linux-x86_64-cpython-38/UnityPyBoost/AnimationClip.o -std=c++20
  x86_64-linux-gnu-g++: error: unrecognized command line option ‘-std=c++20’; did you mean ‘-std=c++2a’?
  /tmp/pip-build-env-jlppwb2m/overlay/lib/python3.8/site-packages/setuptools/command/build_py.py:218: _Warning: Package 'UnityPy.lib.FMOD.Linux.x86_64' is absent from the `packages` configuration.
  !!
  
          ********************************************************************************
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'UnityPy.lib.FMOD.Linux.x86_64' as an importable package[^1],
          but it is absent from setuptools' `packages` configuration.
  
          This leads to an ambiguous overall configuration. If you want to distribute this
          package, please make sure that 'UnityPy.lib.FMOD.Linux.x86_64' is explicitly added
          to the `packages` configuration field.
  
          Alternatively, you can also rely on setuptools' discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
  
          You can read more about "package discovery" on setuptools documentation page:
  
          - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html
  
          If you don't want 'UnityPy.lib.FMOD.Linux.x86_64' to be distributed and are
          already explicitly excluding 'UnityPy.lib.FMOD.Linux.x86_64' via
          `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
          you can try to use `exclude_package_data`, or `include-package-data=False` in
          combination with a more fine grained `package-data` configuration.
  
          You can read more about "package data files" on setuptools documentation page:
  
          - https://setuptools.pypa.io/en/latest/userguide/datafiles.html
  
  
          [^1]: For Python, any directory (with suitable naming) can be imported,
                even if it does not contain any `.py` files.
                On the other hand, currently there is no concept of package data
                directory, all directories are treated like packages.
          ********************************************************************************
  
  !!
    check.warn(importable)
  error: command '/usr/bin/x86_64-linux-gnu-g++' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for UnityPy
ERROR: Could not build wheels for UnityPy which use PEP 517 and cannot be installed directly
```

**Bug**
Failed to build wheel for UnityPy.

**To Reproduce**
- System: Ubuntu 24.04 LTS
- Python3: 3.8.2-0ubuntu2
- Python3 pip: 20.0.2-5ubuntu1.10.
- UnityPy: Directly cloned from Github. Commit: b4bc2ce500d702db16fd5c34309adde0fa651762

**Extra**
Seems that the issue only occurred after the `setup.py` was changed recently(Maybe 2 days ago). For your reference, the C++ 20 compilation command in Linux is `-std=c++2a` and the C++ 23 compilation command is `-std=c++2b`.

**Comment by K0lb3** [comment]
``-std=c++20`` is for GCC and Clang 10 and newer.
``std=c++2a`` is for GCC and Clang 9 and earlier.
Source: [C++20 Support in GCC](https://gcc.gnu.org/projects/cxx-status.html#cxx20) and [C++ Support in Clang]

...so....the setup actually has to select the correct one based on the compiler version... 😒

**Comment by LittleYang0531** [comment]
> ``-std=c++20`` is for GCC and Clang 10 and newer.
> ``std=c++2a`` is for GCC and Clang 9 and earlier.
> Source: [C++20 Support in GCC](https://gcc.gnu.org/projects/cxx-status.html#cxx20) and [C++ Support in Clang]
> 
> ...so....the setup actually has to select the correct one based on the compiler version... 😒 

Oh maybe you are right. I just checked the g++ version and it shows 9.3.0.

**Comment by K0lb3** [comment]
Solved with 1.20.3.

**Comment by LittleYang0531** [comment]
Thanks.
